### PR TITLE
first shot

### PR DESCRIPTION
**Pull request by krokrob**
wip new API endpoint for `/demoday` on `www`. @ssaunier please review, @cedricmenteau please tell me what you want in the json

### CODE REVIEW

File: app/controllers/api/v1/batches_controller.rb [hunk 1]
```diff
@@ -6,4 +6,8 @@ def show
   def live
     @batch = Batch.where(live: true).order("slug DESC").first
   end
+
+  def index
+    @batches = Batch.includes(:projects).where.not(projects: { id: nil }).order(:slug)
```
Comment (ssaunier) [inline app/controllers/api/v1/batches_controller.rb]: @krokrob To be sure, add a filter of projects where `starts_at` is in the past.

File: app/views/api/v1/batches/index.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,9 @@
+json.batches do
+  json.array! @batches do |batch|
+    json.batch do
```
Comment (ssaunier) [inline app/views/api/v1/batches/index.json.jbuilder]: You should partialize the `show.json.jbuilder` in a `_batch.json.jbuilder` to be used here and in the show. That way @cedricmenteau  will have almost all he needs.

**Comment by krokrob** [comment]
@ssaunier , @cedricmenteau here is the json at `alumni.lewagon.org/api/v1/batches`:
```
@batches
=> #<Api::Batch:0x007f969f9e1938 @json=[
  {"batch"=>{"id"=>1, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>10, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>11, "trello_inbox_list_id"=>"", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>15, "trello_inbox_list_id"=>"", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>16, "trello_inbox_list_id"=>"", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>18, "trello_inbox_list_id"=>"5592467ea401f87431ccbb7a", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>19, "trello_inbox_list_id"=>"560cfea931029c41d4bcec16", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>2, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>31, "trello_inbox_list_id"=>"", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>20, "trello_inbox_list_id"=>"55ae3cd4c4920b0e95b671ee", "price_cents"=>450000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>23, "trello_inbox_list_id"=>"5628e228993aec9dbf33fc4d", "price_cents"=>490000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>24, "trello_inbox_list_id"=>"563b3803ec31e6ba47633963", "price_cents"=>490000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>41, "trello_inbox_list_id"=>"56f14c7ab2bb74744b7aaf0b", "price_cents"=>592900, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>3, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>25, "trello_inbox_list_id"=>"563b38ab4e6b7cef47d36fa8", "price_cents"=>490000, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>40, "trello_inbox_list_id"=>"56f13eabea2b664bfbd7b7dc", "price_cents"=>1490000, "price_currency"=>"BRL"}},
  {"batch"=>{"id"=>4, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>5, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>6, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>7, "trello_inbox_list_id"=>"", "price_cents"=>300001, "price_currency"=>"EUR"}},
  {"batch"=>{"id"=>8, "trello_inbox_list_id"=>nil, "price_cents"=>0, "price_currency"=>"EUR"}}
]> ```

**Comment by ssaunier** [comment]
(deleted comment not to expose too much information on a public repo)

File: app/controllers/api/v1/batches_controller.rb (hunk 1 above)
Comment (krokrob) [inline app/controllers/api/v1/batches_controller.rb]: je m'occupe de ça

**Comment by krokrob** [comment]
@ssaunier please review

**Pull request merged**